### DISCUSSION TITLE
Erster Start

### DISCUSSION THREAD

**Discussion by patman15** [SukramJ/aiohomematic#221] [Ideas]
Hallo!

Hatte schon eine kurze Unterhaltung mit @danielperna84 über den ersten Start, bei dem ich etwas verwirrt wurde.

Ein paar Beobachtungen:

1. Wenn man nur das Nötigste ausfüllt, hat man am Ende der Installation nur die Zentrale als Gerät und wundert sich was da los ist.
2. Wenn man den JSON Port nicht ausfüllt (ist ja nicht zwingend) hat man keine Namen und
3. wenn man ihn dann nachträglich ausfüllt hilft das nicht mehr.

Ich denke alle drei Punkte sind nicht tragisch, wenn man irgendwie darauf hingewiesen wird. Die Installation per HACS fordert einen  auf einen Neustart zu machen, aber danach die Integration nach der Konfiguration fragt nach keinem solchen bzw. auch nicht nach "neu laden".

JSON-Port sagt nur eingeweihten was (gebe zu hab ihn absichtlich ignoriert). Ich würde einen Hinweise auf die Namensauflösung geben bzw. vielleicht per Default mit 80 aktivieren und für Experten Deaktivieren ermöglichen, dann kann weniger passieren. Damit wäre dann auch (3) erledigt.

Meine paar Cents, zu einem wirklich gelungenem Setup, das mich auf Anhieb begeistert hat. Aktuell halten mich nur mögliche Breaking-Changes von einem Umstieg im Produktivsystem ab (aber vielleicht werde ich mutiger). Falls es was Konkretes zum Testen oder anderweitig beizutragen gibt, bin ich gerne dazu bereit!

Danke für die Bemühungen die HM Integration auf einen aktuellen Stand zu heben!

**Comment by emufan** [comment]
> 1. Wenn man nur das Nötigste ausfüllt, hat man am Ende der Installation nur die Zentrale als Gerät und wundert sich was da los ist.

Was wäre denn Dein Vorschlag? ME lässt sich das ja nicht ändern, denn die einen haben HMIP, die anderen HM und die dritten HMWired. Das sind ja disjunkte Systeme.

> 2\. Wenn man den JSON Port nicht ausfüllt (ist ja nicht zwingend) hat man keine Namen und

War bei mir nicht so und sollte nicht so sein. Denke, die Integration nimmt den Default-Port. 

Wobei es mir am Anfang auch so war, dass ich 2-3 Mal neu starten musste, bis alles da war.

**Reply by patman15** [reply]
> > 1. Wenn man nur das Nötigste ausfüllt, hat man am Ende der Installation nur die Zentrale als Gerät und wundert sich was da los ist.
> Was wäre denn Dein Vorschlag? ME lässt sich das ja nicht ändern, denn die einen haben HMIP, die anderen HM und die dritten HMWired. Das sind ja disjunkte Systeme.
> 
Vielleicht hab ich das schlecht formuliert. Was ich meinte ist, dass ein Hinweis am Ende des Konfigurationprozesses, dass man die Integration einmal neu laden soll hilfreich wäre. Nach der Installation durch HACS werde ich zum Neustart aufgefordert, nach der Konfiguration der Integration, fehlen einfach die Geräte. Lässt sich aber einfach durch Neuladen reparieren.

> > 2. Wenn man den JSON Port nicht ausfüllt (ist ja nicht zwingend) hat man keine Namen und
> War bei mir nicht so und sollte nicht so sein. Denke, die Integration nimmt den Default-Port.
> 
Stimmt, hab's heute nochmal probiert. Interessanterweise ging es heute auch ohne Angabe und trotzdem mit Namen. Seltsam.

**Comment by danielperna84** [comment]
1. Ich habe die [Installationsanleitung](https://github.com/danielperna84/custom_homematic/wiki/Installation) mal um einen Absatz ergänzt der das erwähnt.
2. Der Default-Port ist 80. Solange der erreichbar ist sollte das eigentlich gehen. 
3. Da muss man sich dann (vorerst) glaube ich [hiermit](https://github.com/danielperna84/hahomematic/wiki/Howto's) behelfen.

Ich persönlich weiß nicht, ob man so einen extra Dialog anzeigen kann. Grundsätzlich sollte das aber eigentlich auch ohne das nochmalige Laden funktionieren. Warum das gerade nicht so ist weiß ich nicht. Intuitiv ist es so tatsächlich nicht.

80 ist wie gesagt schon default. Die Erläuterung dazu kommt sowieso in die Doku. Die gibt es halt noch nicht. Kümmer' ich mich aber bald drum (hoffentlich).

**Reply by SukramJ** [reply]
Port 443 ist default wenn TLS ausgewählt wurde!

**Reply by danielperna84** [reply]
Ah, ja, danke für den Hinweis. 👍
